### ISSUE TITLE
Directly pasting image into a post when composing

### ISSUE THREAD

**Issue by HankG** [diaspora/diaspora#7878]
A feature I use on a lot of places which I can't do in Diaspora right now is copy/pasting an image from the clipboard into the composition area and have it become part of the post automatically.  In the current Diaspora a user needs to save it to the disk then add upload it to attach it to the post...

**Comment by HankG** [comment]
It turns out that the FineUploader we are already using supports this option.  You just have to set up the paste options and it works.  The big question is if we want the paste-focus to be on the text area or the entire page.  I coded it up against the text area first.  Thoughts?

**Comment by Flaburgan** [comment]
It has to be the textarea as you can have the publisher open several times in the page to comment.

**Comment by Flaburgan** [comment]
We also have the drag and drop feature to upload an image. It currently only works on the camera icon, maybe you can extend it to works on the whole textarea.

**Comment by HankG** [comment]
I'll look into that.  Right now drag and drop into the text area drops the file path to text box.

**Comment by HankG** [comment]
@Flaburgan yes I got your suggestion working but had to bring in a bit more of the FineUploader library (we only brought in the core before). It's only a net couple hundred K if I'm reading it correctly.  I've tested this manually and am looking at if I need to make an update to the Jasmine tests.

**Comment by HankG** [comment]
I'm seeing an inconsistency between the mobile and desktop calls into the photos service.  This goes back to the earliest versions of both.  The mobile one adds the aspects fields to the POST arguments while the desktop one does not when calling the photos upload service.  I'm going to do some experimenting but does this discrepancy make any sense to anybody?

**Comment by HankG** [comment]
I have a PR that captures the original request, plus the enhanced drag and drop requested by @Flaburgan , and with the change to more consistently call into the file uploaded between the two. https://github.com/diaspora/diaspora/pull/7883

**Comment by Siedlerchr** [comment]
Drag and Drop  Support as an intermediate step would be also nice.

**Comment by HankG** [comment]
@Siedlerchr the PR for this does drag and drop as well. Drag and drop already worked if you put it on top of the camera icon.  In the current version it works if you drop it into the text field or the thumbnail bar too.
